code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -189,6 +189,10 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 			var/mob/living/M = loc
 			if (!nomessage)
 				to_chat(M, "<span class='notice'>Your [name] goes out.</span>")
+			if(ishuman(M))
+				var/mob/living/carbon/human/H=M
+				if(H.a_intent!=INTENT_HARM && !H.incapacitated())
```

Comment:
```suggestion
				if(H.a_intent==INTENT_HELP && !H.incapacitated())
```
Personally, I think all intents but help intent are pretty much used for combat often. Disarm's for shoving, grabbing's for the impossible to escape nonsense, and harm intent is... harm. It probably won't happen often, but smoker characters in the middle of combat probably might get inconvenienced by one of their hands suddenly getting a cig in hand.

This is my personal recommendation.